Merge daemon defaults with user-supplied jvmargs (#20054)
gradle/gradle: subprojects/docs/src/docs/userguide/running-builds/build_environment.adoc

`org.gradle.jvmargs=(JVM arguments)`::

> Specifies the JVM arguments used for the Gradle Daemon. The setting is particularly useful for <<#sec:configuring_jvm_memory,configuring JVM memory settings>> for build performance. This does not affect the JVM settings for the Gradle client VM.

Author: I've pushed some improvements to the doc. PTAL: 438a568

> Specifies the JVM arguments used for the Gradle Daemon. The setting is particularly useful for <<#sec:configuring_jvm_memory,configuring JVM memory settings>> for build performance. This does not affect the JVM settings for the Gradle client VM.

Reviewer: I made the wording better (in my opinion), PTAL: 24a222e8a36f751b598c0f60150af0d27bf9c008

_A reasonable default is derived from your environment (`JAVA_HOME` or the path to `java`) if the setting is unspecified._

`org.gradle.jvmargs=(JVM arguments)`::

> Specifies JVM arguments to be used for the Gradle Daemon. The setting is particularly useful for <<#sec:configuring_jvm_memory,configuring JVM memory settings>> for build performance. This does not affect the JVM settings for the Gradle client VM.
> Supplied values are merged with the defaults, with special handling for <<#sec:configuring_jvm_memory,JVM memory settings>> (see link).
> _The default values are `-Xms256m -Xmx512m -XX:MaxMetaspaceSize=256m -XX:+HeapDumpOnOutOfMemoryError`._

Reviewer 2: ```suggestion
Specifies the JVM arguments to use for the Gradle Daemon. The setting is particularly useful for <<#sec:configuring_jvm_memory,configuring JVM memory settings>> for build performance. This does not affect the JVM settings for the Gradle client VM.
Gradle merges supplied values with the defaults, with special handling for <<#sec:configuring_jvm_memory,JVM memory settings>>.
_The default value is `-Xms256m -Xmx512m -XX:MaxMetaspaceSize=256m -XX:+HeapDumpOnOutOfMemoryError`._
```
Suggestion: slight reword for docs consistency. I suspect we should specify the default *value* here (not values) because the default value refers to the singular `org.gradle.jvmargs` property, not the plural JVM arguments.

You can adjust JVM options for Gradle in the following ways:

> The `org.gradle.jvmargs` Gradle property specifies arguments for the JVM running the build. The default values are `-Xms256m -Xmx512m -XX:MaxMetaspaceSize=256m -XX:+HeapDumpOnOutOfMemoryError`.

Reviewer 2: ```suggestion
The `org.gradle.jvmargs` Gradle property specifies arguments for the JVM running the build. The default value is `-Xms256m -Xmx512m -XX:MaxMetaspaceSize=256m -XX:+HeapDumpOnOutOfMemoryError`.
```
Same suggestion as above -- I *think* this ought to be singular.

> JVM arguments provided via `org.gradle.jvmargs` are combined with the defaults, with special handling for `-Xms` and `-Xmx`: setting either these values will remove the default for the other.
> This avoids the case where a default value is incompatible with a user-supplied value, such as having a minimum memory setting higher than the maximum.

Reviewer 2: ```suggestion
Gradle combines JVM arguments provided via `org.gradle.jvmargs` with the default JVM arguments. Setting either `-Xms` and `-Xmx` removes the default for the other.
This avoids incompatible combinations of default and user-supplied values, such as a minimum memory setting higher than the maximum.
```
Suggestion: active voice rephrase to make the first sentence more readable. Trimmed some words from the second sentence for the same reason.
